src/chocolatey.tests.integration/infrastructure/cryptography/CryptoHashProviderSpecs.cs

```diff
@@ -63,5 +63,55 @@ public void Should_provide_the_correct_hash_based_on_a_checksum()
                 _result.Should().Be(expected);
             }
         }
+
+        public class When_HashProvider_is_given_a_large_file_to_hash : CryptoHashProviderSpecsBase
+        {
+            private string _result;
+            private string _filePath;
+
+            public override void Context()
+            {
+                base.Context();
+
+                // We need to create a large, 2.5GB file on disk, so that we can run
+                // the ComputeFileHash method against that file, and then test the
+                // generated hash of the file to make sure that the method is working
+                // as expected. It takes around 15-20 seconds to create the file on
+                // disk, which is "fine", and is better than adding a large 2.5 GB
+                // file directly into the repository.
+                var sizeInMb = 2500;
+                const int blockSize = 1024 * 8;
+                const int blocksPerMb = (1024 * 1024) / blockSize;
+                byte[] data = new byte[blockSize];
+                Random rng = new Random(13);
+                _filePath = FileSystem.CombinePaths(ContextDirectory, "large-file.txt");
+
+                using (FileStream stream = File.OpenWrite(_filePath))

```

Comment:
This will keep appending the file if it already exist, causing the file to become larger.

Would be better to use `.Create` instead.